### PR TITLE
Added 'getting started' to documentation

### PR DESCRIPTION
**Pull request by cgoldammer**
I followed the style in the DataFrames documentation. Thus, it's intended as a walk-through of the basic features without links to the more detailed documentation. For code blocks, the display differs from the rest of the documentation because I include the ">julia" from the REPL. This is because I want to include both code and output.

Very much looking forward to feedback. Happy to add more or do extensive editing as needed.

### CODE REVIEW

File: doc/source/starting.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+.. code-block:: julia
+
+    julia> using Distributions
+    julia> srand(123) # Setting the seed
+
+We first create a ``normal`` distribution and then obtain samples using ``rand``:
+
+.. code-block:: julia
+
+    julia> normal = Normal()
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: I think using case only to distinguish an object from a type is a little confusing for a beginner.

File: doc/source/starting.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    Normal(μ=0.0, σ=1.0)
+
+    julia> x = rand(normal, 100)
+    100-element Array{Float64,1}:
+      0.376264
+     -0.405272
+     ...
+
+You can easily obtain a pdf, cdf, percentile, and many other functions for a distribution. For instance, the median (50th percentile) and the 95th percentile for the standard-normal distribution are
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: Probably don't want the "a" here given the rest of this sentence.

File: doc/source/starting.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+.. code-block:: julia
+
+    julia> quantile(Normal(), [0.5, 0.95])
+    2-element Array{Float64,1}:
+     0.0
+     1.64485
+
+The normal distribution is parameterized by a mean (location) and standard deviation (scale). To draw random samples from a normal distribution with mean 1 and standard deviation 2:
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: I'd avoid mentioning location and scale.

I'd add a second clause in the second sentence to make it a complete sentence, e.g. "standard deviation 2, you can do something like:"

File: doc/source/starting.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+-----------
+
+The package contains a large number of additional distributions of four main types:
+
+* ``Univariate``
+* ``Truncated``
+* ``Multivariate``
+* ``Matrixvariate``
+
+The ``Univariate`` random variables split further into ``Discrete`` and ``Continuous``.
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: All the distributions allow this split, not just Univariate.

File: doc/source/starting.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+For instance, you can define the following distributions (among many others):
+
+.. code-block:: julia
+
+    julia> Binomial(p) # Discrete univariate
+    julia> Cauchy(u, b)  # Continuous univariate
+    julia> TruncatedNormal(Normal(mu, sigma), l, u) # Truncated
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: I don't think TruncatedNormal is a type. I think you mean Truncated.

File: doc/source/starting.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+It is often useful to approximate an empirical distribution with a theoretical distribution. As an example, we can use the array ``x`` we created above and ask which normal distribution best describes it:
+
+.. code-block:: julia
+
+    julia> fit(Normal, x)
+    Normal(μ=0.036692077201688635, σ=1.1228280164716382)
+
+Since ``x`` is a random draw from ``Normal``, it's easy to check that the fitted values are sensible. Indeed, the estimates :math:`[0.04, 1.12]` are close to the true values of :math:`[0.0, 1.0]` that we used to generate ``x``.
+
+Create Mixture Models
```
Comment (johnmyleswhite) [inline doc/source/starting.rst]: Let's avoid documenting mixture models since that interface might need to change soon.

**Comment by johnmyleswhite** [comment]
This is great. Thanks so much for working on this.

File: doc/source/starting.rst (hunk 1 above)
Comment (cgoldammer) [inline doc/source/starting.rst]: What would be easier? Omitting line 22 and just saying `x=rand(Normal(), 100)`? Or would you want me to use different notation?

**Comment by cgoldammer** [comment]
This all sounds great. I'll get these edits in later today.

File: doc/source/starting.rst (hunk 1 above)
Comment (johnmyleswhite) [inline doc/source/starting.rst]: I often use `d` for a distribution object.

**Comment by cgoldammer** [comment]
A quick note: Upon submitting my pull request, Travis shows that code coverage decreased by 10%. Within 1/2 hour or so, this gets updated to the correct change (0%).

Very likely a problem with Travis, not this package. Happy to forward this issue to another location.

**Comment by simonster** [comment]
This might be because we run code coverage on both Julia release and nightly and they give different results for reasons unrelated to the package.

**Comment by cgoldammer** [comment]
@simonster Good to know!

**Comment by simonbyrne** [comment]
Fantastic, thanks for doing this!

**Pull request merged**